components/user_settings/display/manage_languages/manage_languages.tsx

```diff
@@ -1,144 +1,176 @@
             this.setState({openMenu: true});
         }
-    }
+    };
 
-    setLanguage = (selectedOption) => {
+    setLanguage = (selectedOption: any) => {
         this.setState({
             locale: selectedOption.value,
-            selectedOption,
+            selectedOption
         });
-    }
+    };
 
     changeLanguage = () => {
         if (this.props.user.locale === this.state.locale) {
             this.props.updateSection('');
         } else {
             this.submitUser({
                 ...this.props.user,
-                locale: this.state.locale,
+                locale: this.state.locale
             });
         }
-    }
+    };
 
-    submitUser = (user) => {
+    submitUser = (user: UserProfile) => {
         this.setState({isSaving: true});
 
-        this.props.actions.updateMe(user).
-            then(({data, error: err}) => {
-                if (data) {
-                    // Do nothing since changing the locale essentially refreshes the page
-                } else if (err) {
-                    let serverError;
-                    if (err.message) {
-                        serverError = err.message;
-                    } else {
-                        serverError = err;
-                    }
-                    this.setState({serverError, isSaving: false});
+        this.props.actions.updateMe(user).then((res) => {
+            if ('data' in res) {
+                // Do nothing since changing the locale essentially refreshes the page
+            } else if ('error' in res) {
+                let serverError;
+                const {error} = res;
+                if (error instanceof Error) {
+                    serverError = error.message;
+                } else {
+                    serverError = error;
                 }
-            });
-    }
+                this.setState({serverError, isSaving: false});
+            }
+        });
+    };
 
     handleMenuClose = () => {
-        document.querySelector('.modal-body').classList.remove('no-scroll');
+        const modalBody = document.querySelector('.modal-body');
+        if (modalBody) {
+            modalBody.classList.remove('no-scroll');
+        }
         this.setState({openMenu: false});
-    }
+    };
 
     handleMenuOpen = () => {
-        document.querySelector('.modal-body').classList.add('no-scroll');
+        const modalBody = document.querySelector('.modal-body');
+        if (modalBody) {
+            modalBody.classList.add('no-scroll');
+        }
         this.setState({openMenu: true});
-    }
+    };
 
     render() {
         let serverError;
         if (this.state.serverError) {
-            serverError = <label className='has-error'>{this.state.serverError}</label>;
+            serverError = (
+                <label className='has-error'>{this.state.serverError}</label>
+            );
         }
 
-        const options = [];
-        const locales = I18n.getLanguages();
+        const options: object[] = [];
```

Comment:
You can type this as `{value: string, label: string}`

Reply:
resolved